test/ServiceProxyTest.cpp

```diff
@@ -0,0 +1,255 @@
+        .WillOnce(Return("Maximum power cap"));
+    EXPECT_CALL(*m_bus_reply, read_integer())
+        .WillOnce(Return(1))
+        .WillOnce(Return(2));
+    EXPECT_CALL(*m_bus, make_call_message("PlatformGetControlInfo"))
+        .WillOnce(Return(m_bus_message));
+    std::vector<std::string> input_names = {"frequency", "power"};
+    EXPECT_CALL(*m_bus_message, append_strings(input_names))
+        .Times(1);
+    std::shared_ptr<SDBusMessage> bus_message_ptr = m_bus_message;
+    EXPECT_CALL(*m_bus, call_method(bus_message_ptr))
+        .WillOnce(Return(m_bus_reply));
+    std::vector<control_info_s> info_actual = m_proxy->platform_get_control_info(input_names);
+
+    std::vector<control_info_s> info_expect = {{"frequency",
+                                               "Maximum cpu frequency",
+                                                1},
+                                              {"power",
+                                               "Maximum power cap",
+                                                2}};
+    ASSERT_EQ(info_actual.size(), info_expect.size());
+    for (unsigned info_idx = 0; info_idx != info_actual.size(); ++info_idx) {
+        check_control_info(info_expect.at(info_idx), info_actual.at(info_idx));
+    }
+}
+
+void ServiceProxyTest::check_control_info(const control_info_s &actual,
+                                          const control_info_s &expect) const
+{
+    EXPECT_EQ(actual.name, expect.name);
+    EXPECT_EQ(actual.description, expect.description);
+    EXPECT_EQ(actual.domain, expect.domain);
+}
+
+TEST_F(ServiceProxyTest, platform_open_session)
+{
+    EXPECT_CALL(*m_bus, call_method("PlatformOpenSession"));
+    m_proxy->platform_open_session();
+}
+
+TEST_F(ServiceProxyTest, platform_close_session)
+{
+    EXPECT_CALL(*m_bus, call_method("PlatformCloseSession"));
+    m_proxy->platform_close_session();
+}
+
+TEST_F(ServiceProxyTest, platform_start_batch)
+{
+    std::vector<struct geopm_request_s> signal_config;
+    std::vector<struct geopm_request_s> control_config;
+    int server_pid = -1;
+    std::string server_key;
+    GEOPM_EXPECT_THROW_MESSAGE(m_proxy->platform_start_batch(signal_config,
+                                                             control_config,
+                                                             server_pid,
+                                                             server_key),
+                               GEOPM_ERROR_NOT_IMPLEMENTED,
+                               "platform_start_batch");
+}
+
+TEST_F(ServiceProxyTest, platform_stop_batch)
+{
+    GEOPM_EXPECT_THROW_MESSAGE(m_proxy->platform_stop_batch(100),
+                               GEOPM_ERROR_NOT_IMPLEMENTED,
+                               "platform_stop_batch");
+}
+
+TEST_F(ServiceProxyTest, platform_read_signal)
+{
+    EXPECT_CALL(*m_bus, call_method("PlatformReadSignal", "instructions", 1, 2));
+    m_proxy->platform_read_signal("instructions", 1, 2);
```

Comment:
didn't mock the call to read the return message or return the value here... not quite done yet, will seg fault when run.